### ISSUE TITLE
Feature: Goal posts visulizations

### ISSUE THREAD

**Issue by antequeradani** [andrewRowlinson/mplsoccer#100]

**Comment by daryl-g** [comment]
<img width="2400" height="1632" alt="Image" src="https://github.com/user-attachments/assets/2c6bd412-e277-48bf-9743-2d5b20a2344d" />

Not sure if this is what you are looking for (basically a goalmouth visualisation), but I just want to leave this code here if anyone is make a similar visualisation and since I'm unsure if @andrewRowlinson still accepts new pull request.

Basically, this code uses the y-coordinate from any provider's data on the x-axis, and the z-coordinate on the y-axis.
```py
## Set up goalmouth
fig, ax = plt.subplots(figsize=(10, 6))

### These coordinates are for Fotmob's data, please adjust these limits and coordinates based on the data source
ax.axis("off")
ax.spines["top"].set_visible(False)
ax.spines["right"].set_visible(False)
ax.spines["bottom"].set_visible(False)
ax.spines["left"].set_visible(False)
ax.set_xlim(39, 28.9)
ax.set_ylim(-1, 5)

## Draw goalmouth
### Goalmouth
ax.plot([30.5, 37.7], [2.5, 2.5], color="white", linewidth=3, zorder=1)
ax.plot([30.5, 30.5], [0, 2.5], color="white", linewidth=3, zorder=1)
ax.plot([37.7, 37.7], [0, 2.5], color="white", linewidth=3, zorder=1)
### Bottom line
ax.plot([28.9, 39], [0, 0], color="white", linewidth=2.5, alpha=1, zorder=0)

### Zone dividers (optional)
ax.plot(
    [35.4, 35.4],
    [0, 2.5],
    color="white",
    linewidth=1,
    linestyle="--",
    alpha=0.5,
    zorder=0,
)
ax.plot(
    [32.9, 32.9],
    [0, 2.5],
    color="white",
    linewidth=1,
    linestyle="--",
    alpha=0.5,
    zorder=0,
)
ax.plot(
    [30.5, 37.7],
    [1.15, 1.15],
    color="white",
    linewidth=1,
    linestyle="--",
    alpha=0.5,
    zorder=0,
)
```
